File: common.addons/src/main/java/com/vp/commonaddons/di/SharePreferencesModule.java
package com.vp.commonaddons.di;

import com.google.gson.Gson;
import com.google.gson.GsonBuilder;

import dagger.Module;
import dagger.Provides;

@Module
public class SharePreferencesModule {

    @Provides
    Gson providesGson() {
        return new GsonBuilder().create();
    }
}

File: common.addons/src/main/java/com/vp/commonaddons/di/SharePreferencesModule.kt
package com.vp.commonaddons.di

import com.google.gson.Gson
import com.google.gson.GsonBuilder
import dagger.Module
import dagger.Provides

@Module
class SharePreferencesModule {
    @Provides
    fun providesGson(): Gson {
        return GsonBuilder().create()
    }
}